tests/integration/components/common/busy-model-wrapper-test.js

```diff
@@ -1,25 +1,48 @@
 import { moduleForComponent, test } from 'ember-qunit';
 import hbs from 'htmlbars-inline-precompile';
+import pageObject from 'ember-cli-page-object';
+import busyModelWrapper from 'code-corps-ember/tests/integration/components/common/busy-model-wrapper-test';
+import { set } from '@ember/object';
+
+let page = pageObject.create(busyModelWrapper);
+
+function renderPage() {
+  page.render(hbs`
+  {{common/busy-model-wrapper
+    model=model
+  }}`);
+}
 
 moduleForComponent('common/busy-model-wrapper', 'Integration | Component | common/busy model wrapper', {
-  integration: true
+  integration: true,
+  beforeEach() {
+    page.setContext(this);
+  },
+  afterEach() {
+    page.removeContext();
+  }
 });
 
-test('it renders', function(assert) {
+test('it shows "saving" if model is saving, but not deleted', function(assert) {
+  let model = { isSaving: true, isDeleted: false };
+  set(this, 'model', model);
+  renderPage();
 
-  // Set any properties with this.set('myProperty', 'value');
-  // Handle any actions with this.on('myAction', function(val) { ... });
+  assert.equal(page.text, 'Saving...', 'the model is saving');
 
-  this.render(hbs`{{common/busy-model-wrapper}}`);
+});
 
-  assert.equal(this.$().text().trim(), '');
+test('it shows "Deleting" if model is saving and is deleted', function(assert) {
+  let model = { isSaving: true, isDeleted: true };
+  set(this, 'model', model);
+  renderPage();
+
+  assert.equal(page.text, 'Deleting...', 'the model is deleting');
+});
 
-  // Template block usage:
-  this.render(hbs`
-    {{#common/busy-model-wrapper}}
-      template block text
-    {{/common/busy-model-wrapper}}
-  `);
+test('it shows yeild if model is neither saving or deleted', function(assert) {
+  let model = { isSaving: false, isDeleted: false };
+  set(this, 'model', model);
 
-  assert.equal(this.$().text().trim(), 'template block text');
+  assert.equal(page.text, 'Foo', 'the model returns block');
 });
```

Comment:
Have the first two tests passing, by setting the model, however, I'm unsure how to get the this one passing without something to yield. Nikola mentioned earlier that I could set my renderPage to page.render(hbs{{#busy-model-wrapper model=model}}Foo{{/busy-model-wrapper}}) but I've tried this both in my initial renderPage() function and in the test itself and haven't gotten anywhere with it. The error I get is

Error: Element not found.

PageObject: 'page.text'
Selector: ':first'

I got this on both of my tests prior, but only because I'd messed up the boolean values onSaving and onDeleting. I fixed that, but I'm not sure what this would need to accept that neither onSaving or onDeleting are false.

Reply:
You won't want to use `renderPage` for this since that's just a helper function.

In this case you would do:

```js
page.render(hbs`
  {{#busy-model-wrapper model=model}}
    Foo
  {{/busy-model-wrapper}}
`);
```

inside of your test.